llvm/lib/Analysis/LoopCacheAnalysis.cpp

```diff
@@ -338,14 +340,18 @@ CacheCostTy IndexedReference::computeRefCost(const Loop &L,
   assert(RefCost && "Expecting a valid RefCost");
 
   // Attempt to fold RefCost into a constant.
+  // CacheCostTy is a signed integer, but the tripcount value can be large
+  // and may not fit, so saturate/limit the value to the maximum signed
+  // integer value.
   if (auto ConstantCost = dyn_cast<SCEVConstant>(RefCost))
-    return ConstantCost->getValue()->getZExtValue();
+    return ConstantCost->getValue()->getLimitedValue(
```

Comment:
Is this still needed or sufficient to let wrapping handle by InstructionCost?

Reply:
Yes, this is unfortunately still needed. Helper `getZExtValue()` returns an uint64_t and checks if the value fits in this number of bits.